gfx/video_thread_wrapper.c

```diff
@@ -713,26 +724,17 @@ static void video_thread_set_viewport(void *data, unsigned width,
       unsigned height, bool force_full, bool video_allow_rotate)
 {
    thread_video_t *thr = (thread_video_t*)data;
-   if (!thr)
-      return;
 
-   slock_lock(thr->lock);
```

Comment:
Are you certain it is safe to remove these locks...?

Reply:
Yes. We never set the viewport at the driver's thread. Some drivers (vulkan for example), may set the viewport at viewport_info, but that never goes through video_thread_set_viewport, so the lock wouldn't be acquired anyway.

Enforcing locks at viewport_info will be costly as it will happen at every frame updated, thus, I would suggest only doing that if a problem shows up, which probably won't, as the original code didn't acquire the lock for viewport_info.